docs/develop/develop-overview.md

```diff
@@ -0,0 +1,189 @@
+
+The following image shows an add-in command in the ribbon, a task pane to the right of the document, and a dialog box floating over the document.
+
+![An image that shows add-in commands on the ribbon, a task pane, and a dialog box in an Office document](../images/overview-with-app-interface-elements.png)
+
+For more information about extending the Office UI, see [Design Office Add-ins](../design/add-in-design.md).
+
+### Interacting with content in an Office document
+
+An Office Add-in can use the Office JavaScript APIs to interact with content in the Office document where the add-in is running. 
+
+#### Accessing the Office JavaScript API library
+
+The Office JavaScript API library can be accessed via the Office.js content delivery network (CDN) at: `https://appsforoffice.microsoft.com/lib/1/hosted/Office.js`. To use Office JavaScript APIs within any of your add-in's web pages, you must reference the CDN in a `<script>` tag in the `<head>` tag of the page.
+
+```html
+<head>
+    ...
+    <script src="https://appsforoffice.microsoft.com/lib/1/hosted/Office.js" type="text/javascript"></script>
+</head>
+```
+
+> [!NOTE]
+> To use preview APIs, reference the preview version of the Office JavaScript library on the CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+
+#### API object models
+
+The Office JavaScript APIs include two distinct object models:
+
+- **Host-specific** APIs (introduced with Office 2016) provide strongly-typed objects that can be used to interact with objects that are native to a specific Office application. For example, you can use the Excel JavaScript APIs to access worksheets, ranges, tables, charts, and more. Host-specific APIs are currently available for [Excel](../reference/overview/excel-add-ins-reference-overview.md), [Word](../reference/overview/word-add-ins-reference-overview.md), [OneNote](../reference/overview/onenote-add-ins-javascript-reference.md), and [PowerPoint](..//reference/overview/powerpoint-add-ins-reference-overview.md). This object model uses [promises](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+
+- **Common** APIs (introduced with Office 2013) can be used to access features such as UI, dialogs, and client settings that are common across multiple types of Office applications. This object model uses [callbacks](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function). For details about the Common API object model, which includes APIs for interacting with Outlook, see [Office JavaScript API object model](office-javascript-api-object-model.md).
+
+> [!NOTE]
+> Excel Custom functions run within a unique runtime that prioritizes execution of calculations, and therefore uses a slightly different programming model. For details, see [Custom functions architecture](../excel/custom-functions-architecture.md).
```

Comment:
CFs seem different enough to break them out into a third section, since they don't use the Office-js library.

Reply:
Valid suggestion, but I'm going to leave this as a note for now, pending upcoming changes to CF API architecture.